test/extended/openstack/servergroup.go

```diff
@@ -113,3 +147,27 @@ func serverGroupIDsFromName(client *gophercloud.ServiceClient, name string) ([]s
 
 	return IDs, nil
 }
+
+type installConfig struct {
+	ControlPlane struct {
+		Platform struct {
+			OpenStack struct {
+				ServerGroupPolicy string `yaml:"serverGroupPolicy"`
+			} `yaml:"openstack"`
+		} `yaml:"platform"`
+	} `yaml:"controlPlane"`
+}
+
+func installConfigFromCluster(client clientcorev1.ConfigMapsGetter) (installConfig, error) {
+	const installConfigName = "cluster-config-v1"
+
+	cm, err := client.ConfigMaps("kube-system").Get(context.Background(), installConfigName, metav1.GetOptions{})
+	if err != nil {
+		return installConfig{}, err
+	}
+
+	var ic installConfig
+	yaml.Unmarshal([]byte(cm.Data["install-config"]), &ic)
```

Comment:
add error checking here

Reply:
```Go
var ic installConfig
err = yaml.Unmarshal([]byte(cm.Data["install-config"]), &ic)
return ic, err
```
https://pkg.go.dev/gopkg.in/yaml.v2#Unmarshal

Reply:
Done